### ISSUE TITLE
can not fetch related node via custom query

### ISSUE THREAD

**Issue by wavemoroc001** [spring-projects/spring-data-neo4j#2582]
Hi there, I want to query with my custom query but it always get null on related node eg. organizer.

<img width="945" alt="Screen Shot 2565-08-16 at 14 55 43" src="https://user-images.githubusercontent.com/56395197/184827764-89ce9e96-8a30-49e5-81bd-8b719af3e42f.png">

This picture show that my event node should not null in organizer object

![Screen Shot 2565-08-16 at 14 57 37](https://user-images.githubusercontent.com/56395197/184828105-25eb4fc6-81a7-4d86-b347-efe34a6b434c.png)

Event Repository  
```java 
   @Query(value = "Optional Match p1=(m1:Member)-[r1:REGISTER]->(e1:Event)  \n" +
            "With e1 as e1,count(r1) as num\n" +
            "with collect({event:e1,num:num}) as events\n" +
            "\n" +
            "optional match p2=(m2:Member)-[r2:BOOKMARK]->(e2:Event)\n" +
            "where NOT e2 IN events\n" +
            "with e2 as e2, count(r2) as num2, events as events\n" +
            "with collect({event:e2,num:num2}) as el1, events as el2\n" +
            "with el1 + el2 as events\n" +
            "with  apoc.coll.sortMaps(events,\"num\") as events\n" +
            "UNWIND events as event\n" +
            "   with collect(event.event) as events\n" +
            "return events\n" +
            "SKIP $skip LIMIT $limit\n",
            countQuery = "Match p1=(m1:Member)-[r1:REGISTER]->(e1:Event)  \n" +
                    "With e1 as e1,count(r1) as num\n" +
                    "with collect({event:e1,num:num}) as events\n" +
                    "\n" +
                    "Match p2=(m2:Member)-[r2:BOOKMARK]->(e2:Event)\n" +
                    "where NOT e2 IN events\n" +
                    "with e2 as e2, count(r2) as num2, events as events\n" +
                    "with collect({event:e2,num:num2}) as el1, events as el2\n" +
                    "with el1 + el2 as events\n" +
                    "with  apoc.coll.sortMaps(events,\"num\") as events\n" +
                    "UNWIND events as event\n" +
                    "return count(event.event)\n" +
                    "\n")
    Page<Event> getEventsByRegisterAndBookmarkRelationship(Pageable pageable);
```

Event Models
```java
@Node
@NoArgsConstructor
@Getter
@Setter
public class Event {
    @Id
    @GeneratedValue(UUIDStringGenerator.class)
    private String id;

    private String name;

    private String description;

    private String type;

    @DateLong
    private Date startDate;
    @DateLong
    private Date endDate;

    private String coverImageUrl;

    private List<String> postImageUrls;

    private Integer numberOfPeople;

    private String checkInCode;

    // for online event
    private String link;

    @LastModifiedDate
    @DateLong
    private Date updateAt;

    @CreatedDate
    @DateLong
    private Date createAt;

    @Version
    private Long version;

    @DateLong
    private Date deleteAt;

    @Relationship(type = "TAG", direction = Relationship.Direction.OUTGOING)
    private Set<Tag> tags = new HashSet<>();

    @JsonIgnore
    @Relationship(type = "CREATE", direction = Relationship.Direction.INCOMING)
    private Member organizer;

    @JsonIgnore
    @Relationship(type = "FEEDBACK", direction = Relationship.Direction.INCOMING)
    private Set<FeedbackFromRelation> feedbackEvent = new HashSet<>();

    @JsonIgnore
    @Relationship(type = "CHECK_IN",direction = Relationship.Direction.INCOMING)
    private Set<CheckInEventFromRelation> checkInMember = new HashSet<>();

    @JsonIgnore
    @Relationship(type = "REGISTER", direction = Relationship.Direction.INCOMING)
    private Set<RegisterEventFromRelation> registerMember = new HashSet<>();

    // for onsite event
    @Relationship(type = "LOCATE",direction = Relationship.Direction.OUTGOING)
    private Location location;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Event event = (Event) o;

        return id.equals(event.id);
    }

    @Override
    public int hashCode() {
        return id.hashCode();
    }
}
```

Member models
```java
@Node
@NoArgsConstructor
@Getter
@Setter
public class Member  {
    @Id
    private String id;

    private String username;

    private String firstname;

    private String lastname;

    private String profileUrl;

    private String deviceId;

    private String email;

    private String description;

    @LastModifiedDate
    @DateLong
    private Date updateAt;

    @CreatedDate
    @DateLong
    private Date createAt;

    @Version
    private Long version;

    @Relationship(type = "REGISTER", direction = Relationship.Direction.OUTGOING)
    private Set<RegisterEventRelation> registerEvent = new HashSet<>();

    @Relationship(type = "CHECK_IN",direction = Relationship.Direction.OUTGOING)
    private Set<CheckInEventRelation> checkInEvent = new HashSet<>();

    @Relationship(type = "FEEDBACK",direction = Relationship.Direction.OUTGOING)
    private Set<FeedbackRelation> feedbackEvent = new HashSet<>() ;

    @Relationship(type = "PREFER",direction = Relationship.Direction.OUTGOING)
    private Set<Tag> tags = new HashSet<>();

    public Member(String id, String firstname, String lastname, String image, String email,String deviceId) {
        this.id = id;
        this.firstname = firstname;
        this.lastname = lastname;
        this.profileUrl = image;
        this.email = email;
        this.deviceId = deviceId;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Member member = (Member) o;

        return id.equals(member.id);
    }

    @Override
    public int hashCode() {
        return id.hashCode();
    }
}
```

Tag Models
```java
@Node
@NoArgsConstructor
@Getter
@Setter
public class Tag {

    @Id
    @GeneratedValue(UUIDStringGenerator.class)
    private String id;

    private String name;

    @LastModifiedDate
    @DateLong
    private Date updateAt;

    @CreatedDate
    @DateLong
    private Date createAt;

    @Version
    private Long version;

    public Tag(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }
}

```

Cloud you give some advice  to investigate these problems ?

**Comment by michael-simons** [comment]
Please follow the documentation regarding custom queries:

https://docs.spring.io/spring-data/neo4j/docs/current/reference/html/#custom-queries

Especially, you *must* return the relationships, which you aren't doing in your query.

**Comment by wavemoroc001** [comment]
I test on some simple query. the result is not null now (Organizer Object). If I want to get relationship in that object how cloud  i do that ? (eg. REGISTER, FEEDBACK, CHECK_IN)

<img width="846" alt="Screen Shot 2565-08-16 at 20 58 12" src="https://user-images.githubusercontent.com/56395197/184898153-75bc9738-89a4-4d3d-9ff9-26684acc56fa.png">


```java
    @Query(value = "MATCH x=(t:Tag)-[:TAG]-(e:Event)-[:CREATE]-(m:Member) \n" +
            "WHERE t.name In $tags\n" +
            "RETURN e,nodes(x),relationships(x) \n" +
            ":#{orderBy(#pageable)} SKIP $skip LIMIT $limit\n",
            countQuery = "MATCH (t:Tag)-[:TAG]-(e:Event)\n" +
                    "WHERE t.name In $tags \n" +
                    "RETURN COUNT(DISTINCT e.id) \n"
    )
    Page<Event> getEventByTag(List<String> tags, Pageable pageable);
```

**Comment by michael-simons** [comment]
Follow the documentation (https://docs.spring.io/spring-data/neo4j/docs/current/reference/html/#custom-queries.for-relationships.one.record, the key-point is `collect`)

**Comment by wavemoroc001** [comment]
I have tried to add `collect` in my query but still not get any relationships. Do I miss something ? 

Event Repository 

``` java 
    @Query(value = "MATCH x=(t:Tag)-[:TAG]-(e:Event)-[:CREATE]-(m:Member) \n" +
            "WHERE t.name In $tags\n" +
            "RETURN e,collect(nodes(x)),collect(relationships(x)) \n" +
            ":#{orderBy(#pageable)} SKIP $skip LIMIT $limit\n",
            countQuery = "MATCH (t:Tag)-[:TAG]-(e:Event)\n" +
                    "WHERE t.name In $tags \n" +
                    "RETURN COUNT(DISTINCT e.id) \n"
    )
    Page<Event> getEventByTag(List<String> tags, Pageable pageable);
```

**Comment by wavemoroc001** [comment]
Does custom query is support to use projections ?

**Comment by wavemoroc001** [comment]
Does custom query support Projections ? 

if i want to get multiple relationship in `event` node. Cloud you provide some solution ?
I have tried code in this block but it still not get any relationship


```java
    @Query(value = "Match x=(m:Member{id : $memberId})-[:REGISTER]-(e:Event) \n " +
            "Optional Match y=(m)-[:CHECK_IN]-(e)" +
            "Where  timestamp() > e.startDate \n" +
            "return e,nodes(x),relationships(x),nodes(y),relationships(y)  \n" +
            "order by e.startDate desc SKIP $skip LIMIT $limit",
            countQuery = "Match x=(m:Member{id : $memberId})-[:REGISTER]-(e:Event) \n" +
                    "Where  timestamp() > e.startDate \n" +
                    "return count(e)")
    Page<Event> getEventsByRegisterMemberId(String memberId, Pageable pageable);
```

**Comment by meistermeier** [comment]
Sorry that we did not came back to you on this.
Closing this now because it is still not reproducible (anymore?) with current Spring Data Neo4j. It might have been resolved in the mean time (of 1.5 years).
Please don't feel ignored. If there are still any questions regarding this, feel free to re-open this or file a new issue.
